src/fondant/pipeline/pipeline.py

```diff
@@ -142,16 +146,10 @@ def __init__(
         client_kwargs: t.Optional[dict] = None,
         resources: t.Optional[Resources] = None,
     ) -> None:
-        if self._is_custom_component(name_or_path):
-            self.component_dir = Path(name_or_path)
-        else:
-            self.component_dir = self._get_registry_path(str(name_or_path))
-
+        self.name = name
+        self.image = image
+        self.component_spec = component_spec
```

Comment:
I need to check further but maybe we can do without the component_spec here and use the operation_spec along with the image from this point on.